doctrine.rst

```diff
@@ -522,6 +522,33 @@ be able to go to ``/product/1`` to see your new product::
         // in the template, print things with {{ product.name }}
         // return $this->render('product/show.html.twig', ['product' => $product]);
     }
+    
+This in an alternative example using Symfony's autowiring and Dependencyinjection : 
+    
+    // src/Controller/ProductController.php
+    // don't forget the use statement
+    use App\Repository\ProductRepository;
+
+    /**
+     * @Route("/product/{id}", name="product_show")
+     */
+    public function show($id, ProductRepository $productRepository)
+    {
+        $product = $productRepository
+            ->find($id);
+
+        if (!$product) {
+            throw $this->createNotFoundException(
+                'No product found for id '.$id
+            );
+        }
+
+        return new Response('Check out this great product: '.$product->getName());
+
+        // or render a template
+        // in the template, print things with {{ product.name }}
+        // return $this->render('product/show.html.twig', ['product' => $product]);
+    }
```

Comment:
```rst
Another possibility is to use the ``ProductRepository`` injected by the dependency injection container::

    // src/Controller/ProductController.php
    // ...
    use App\Repository\ProductRepository;

    /**
     * @Route("/product/{id}", name="product_show")
     */
    public function show($id, ProductRepository $productRepository)
    {
        $product = $productRepository
            ->find($id);
        
        // ...
    }
```

Reply:
Hello @OskarStark , that's just what i did in the code ?

Reply:
Yes, but it is too long and verbose, can you please use exactly my code block?

Reply:
done